### DISCUSSION TITLE
RAM usage and slow generating time

### DISCUSSION THREAD

**Discussion by generate144** [lllyasviel/Fooocus#2594] [Q&A] [answered]
I'm using a Nvidia gtx 980 and when I run fooocus it uses 90% + of my 32 GB RAM. it also take at least 4 minutes to generate a single photo when using the "Speed" performance setting. windows swap is set according to the Read my. any help?

**Comment by poor7** [comment]
Use Lightning mode. It significantly increases the generation speed. Additionally use the switch `--vae-in-fp16 --unet-in-fp16 --all-in-fp16`  this will further increase speed and reduce memory consumption.

**Reply by generate144** [reply]
Thanks for the reply! I'm using the Lightning mode for now but getting less the great results. how can I use the switch you mentioned?

**Reply by poor7** [reply]
For example, you can edit the ending line of Windows `run.bat` as
`.\python_embeded\python.exe -s Fooocus\entry_with_update.py --vae-in-fp16 --unet-in-fp16 --all-in-fp16`

**Reply by mashb1t** [reply, marked as answer]
@generate144 Sadly the generation time of 3-4min is to be expected with a GTX 980 as well as maybe not running the latest RAM or a HDD, but the GPU could be the bottleneck here.
Using less steps will linearly reduce the time, so lightning with its 4 steps is the fastest you can get with current presets.
Adding the parameters mentioned by @poor7 may reduce quality, but also further increase speed.
Feel free to share your experience / results afterwards.

**Reply by generate144** [reply]
Thanks for the answer, adding the parameters above did slightly reduce the quality but didn't noticeably improve the speed. what upgrades  would you recommend doing for better results?

**Reply by mashb1t** [reply]
@generate144 
Upgrades to your GPU, as GTX 980 is almost 10 years old. I'd suggest getting a (used) RTX 3080 or 4070, but that totally depends on your intentions and / or if you're also a gamer or not.

Other things to consider:
RAM => DDR5
SSD => NVMe M.2 Gen 4, depending on if your mainboard supports this or not

**Comment by ChrissNeo** [comment]
`--attention-split` also can help.
